packages/gtk_vnc.rb

```diff
@@ -3,41 +3,57 @@
-  version '1.0.0'
+  version '1.2.0'
   license 'LGPL-2.1+'
   compatibility 'all'
-  source_url 'https://ftp.gnome.org/pub/GNOME/sources/gtk-vnc/1.0/gtk-vnc-1.0.0.tar.xz'
-  source_sha256 'a81a1f1a79ad4618027628ffac27d3391524c063d9411c7a36a5ec3380e6c080'
+  source_url "https://gitlab.gnome.org/GNOME/gtk-vnc/-/archive/v#{version}/gtk-vnc-v#{version}.tar.bz2"
+  source_sha256 '125af97fdd43a570e62c351b53c26bd82c9cb1146b7f72d9d594e6e07dde78d9'
 
-  binary_url ({
-    aarch64: 'https://downloads.sourceforge.net/project/chromebrew/armv7l/gtk_vnc-1.0.0-chromeos-armv7l.tar.xz',
-     armv7l: 'https://downloads.sourceforge.net/project/chromebrew/armv7l/gtk_vnc-1.0.0-chromeos-armv7l.tar.xz',
-       i686: 'https://downloads.sourceforge.net/project/chromebrew/i686/gtk_vnc-1.0.0-chromeos-i686.tar.xz',
-     x86_64: 'https://downloads.sourceforge.net/project/chromebrew/x86_64/gtk_vnc-1.0.0-chromeos-x86_64.tar.xz',
+  binary_url({
+    aarch64: 'https://downloads.sourceforge.net/project/chromebrew/armv7l/gtk_vnc-1.2.0-chromeos-armv7l.tar.xz',
+     armv7l: 'https://downloads.sourceforge.net/project/chromebrew/armv7l/gtk_vnc-1.2.0-chromeos-armv7l.tar.xz',
+       i686: 'https://downloads.sourceforge.net/project/chromebrew/i686/gtk_vnc-1.2.0-chromeos-i686.tar.xz',
+     x86_64: 'https://downloads.sourceforge.net/project/chromebrew/x86_64/gtk_vnc-1.2.0-chromeos-x86_64.tar.xz'
   })
-  binary_sha256 ({
-    aarch64: '8026b25e42c51f0a124aff3779b9dbb035b331dd2bb0faa60cf10f54a9e77b38',
-     armv7l: '8026b25e42c51f0a124aff3779b9dbb035b331dd2bb0faa60cf10f54a9e77b38',
-       i686: '86b2a47da79c0035ad41efb6ae4a0eb9bee0d931caf54ecbb44dd79183af5ee8',
-     x86_64: 'efa9ae051e5ef07895101f66f93577fd459609b8eb3fd8bb6bc00ee76fd638d3',
+  binary_sha256({
+    aarch64: '62d85534080b53b4b3e0a20c58979db9e550a035414821d210a45fc8965da0ab',
+     armv7l: '62d85534080b53b4b3e0a20c58979db9e550a035414821d210a45fc8965da0ab',
+       i686: '609017642a8be0a5d309b05d07aed5665760eca14af26075c3b719fef3736dc8',
+     x86_64: '5001d552bfa4155bf2c77a1e1a7ee4714125566db0b9ce1e3e3c5e88ace8057d'
   })
 
-  depends_on 'pygtk'
-  depends_on 'libcyrussasl'
+  depends_on 'cairo'
   depends_on 'glib'
-  depends_on 'libgcrypt'
   depends_on 'gobject_introspection'
+  depends_on 'gtk3'
+  depends_on 'libgcrypt'
+  depends_on 'pulseaudio'
 
-  def self.prebuild
-    system "sed -i '226d' meson.build"
+  def self.patch
+    system "sed -i 's,-fstack-protector-strong,-fno-stack-protector,' meson.build"
+    # Source has keycodemapdb repo as submodule
+    @git_dir = 'subprojects/keycodemapdb'
+    @git_hash = '6280c94f306df6a20bbc100ba15a5a81af0366e6'
+    @git_url = 'https://gitlab.com/keycodemap/keycodemapdb.git'
+    FileUtils.rm_rf(@git_dir)
+    FileUtils.mkdir_p(@git_dir)
+    Dir.chdir @git_dir do
+      system 'git init'
+      system "git remote add origin #{@git_url}"
+      system "git fetch --depth 1 origin #{@git_hash}"
+      system 'git checkout FETCH_HEAD'
+    end
```

Comment:
Take advantage of merging https://github.com/skycocker/chromebrew/pull/5598?

Reply:
Possibly... this is for a subproject... and that is I think more complicated.

In theory a `git submodule update --init --recursive` might work. But I'd have to do some testing on that. I don't want to rush #5998 because it might help other packages. Let's get that right.

Reply:
I'm working on a PR to be able to clone submodules/download patches, but I'm waiting on #5598 to merge first so there aren't a bunch of conflicts.

Reply:
FYI now submodules automatically get installed as long as your source_url points to a git location.